salt/engines/libvirt_events.py

```diff
@@ -0,0 +1,617 @@
+# coding=utf-8
+
+'''
+An engine that listens for libvirt events and resend them to the salt event bus.
+
+Example configuration:
+
+.. code-block:: yaml
+
+    engines:
+        - libvirt_events:
+            uri: lxc:///
+            tag_prefix: libvirt
+            filters:
+                - domain/lifecycle
+                - domain/reboot
+                - pool
+
+The default URI is ``qemu:///system`` and the default tag prefix is
+``salt/engines/libvirt_events``.
+
+Filters is a list of event types to relay to the event bus. Items in this list
+can be either one of the main types (``domain``, ``network``, ``pool``,
+``nodedev``, ``secret``), ``all`` or a more precise filter. These can be done
+with values like <main_type>/<subtype>, but better read the engine's code for
+the complete list of sub types. If the filters list contains ``all``, all
+events will be relayed.
+
+Be aware that the list of events increases with libvirt versions, for example
+network events have been added in libvirt 1.2.1.
+
+:depends: libvirt 1.0.0+ python binding
+'''
+
+from __future__ import absolute_import, unicode_literals, print_function
+import logging
+
+# Import salt libs
+import salt.utils.event
+
+# pylint: disable=no-name-in-module,import-error
+from salt.ext.six.moves.urllib.parse import urlparse
+# pylint: enable=no-name-in-module,import-error
+
+log = logging.getLogger(__name__)
+
+
+# pylint: disable=invalid-name
+
+msg = ''
```

Comment:
This type of msg handling and version checking should be done in the `__virtual__()` function.

Reply:
@cbosdo actually @rallytime is right. So just move checks to the `__virtual__` itself:

```python
try:
    import libvirt
except ImportError:
    libvirt = None

def __virtual__():
    '''
    Only load if libvirt python binding is present
    '''
    if libvirt is None:
        msg = 'libvirt module not found'
    elif libvirt.getVersion() < 1000000:
        msg = 'libvirt >= 1.0.0 required'
    else:
        msg = ''

    return bool(libvirt), msg
```
OK, basically copypaste this. :wink:

Reply:
This makes pylint complain on the `if libvirt is None:` line:
[used-before-assignment] Using variable 'libvirt' before assignment

And on the libvirt = None line that you didn't move:
[redefined-outer-name] Redefining name 'libvirt' from outer scope

Reply:
If checks _has_ to be in `__virtual__`, then just mute Lint in this case.